core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -671,33 +698,43 @@ private[zio] trait Schedule_Functions extends Serializable {
    * </pre>
    */
   final def spaced(interval: Duration): Schedule[Any, Int] =
-    forever.delayed(_ + interval)
+    forever.delayed(_ + interval.relative)
 
   /**
    * A schedule that always recurs, increasing delays by summing the
    * preceding two delays (similar to the fibonacci sequence). Returns the
    * current duration between recurrences.
    */
-  final def fibonacci(one: Duration): Schedule[Any, Duration] =
-    delayed(unfold[(Duration, Duration)]((Duration.Zero, one)) {
-      case (a1, a2) => (a2, a1 + a2)
-    }.map(_._1))
+  final def fibonacci(one: Duration): Schedule[Any, Duration] = {
+    val f = delayed[Any, Any](
+      unfold[(Delay, Delay)]((Delay.none, one.relative)) {
+        case (a1, a2) => (a2, a1 + a2)
```

Comment:
Uh oh. The `+` operator here will build `Delay` of increasing size, and eventually consume all memory (!). We will have to unfold using `Duration` and map to `Delay`.